pkg/agent/cniserver/pod_configuration_test.go

```diff
@@ -0,0 +1,464 @@
+
+			flowInstalled := false
+
+			ifConfig := interfacestore.InterfaceConfig{
+				InterfaceName: podIfName,
+				ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{
+					PodNamespace: testPodNamespace,
+					PodName:      testPodNameA,
+					ContainerID:  podInfraContainerID,
+				},
+				OVSPortConfig: &interfacestore.OVSPortConfig{
+					PortUUID: "test-port-uuid",
+				},
+				IPs: podIPs,
+				MAC: podMac,
+			}
+
+			if tc.ofPortAssigned {
+				ifConfig.OVSPortConfig.OFPort = int32(1)
+			}
+
+			if tc.podIfaceIsCached {
+				configurator.ifaceStore.AddInterface(&ifConfig)
+			}
+
+			if tc.installFlow {
+				fakeOFClient.EXPECT().InstallPodFlows(podIfName, podIPs, podMac, portStatusMsg.Desc.PortNo, uint16(0), nil).Times(1).Return(tc.installOpenFlowErr)
+				if tc.installOpenFlowErr == nil {
+					flowInstalled = true
+				}
+			}
+
+			err := configurator.updateUnreadyPod(podIfName)
+			if tc.expErr == "" {
+				require.NoError(t, err)
+			} else {
+				require.EqualError(t, err, tc.expErr)
+			}
+
+			if flowInstalled {
+				assert.True(t, waiter.waitUntil(5*time.Second))
+			}
+
+			if tc.expEvent != "" {
+				actEvt := <-testClients.recorder.Events
+				require.Equal(t, tc.expEvent, actEvt)
+			}
+		})
+	}
+}
+
+func TestProcessNextWorkItem(t *testing.T) {
+	mockRetryInterval(t)
+
+	for _, tc := range []struct {
+		name               string
+		installOpenFlowErr error
+		expEvent           string
+		expRequeue         bool
+	}{
+		{
+			name:               "failed to install OpenFlow entries for updated Port",
+			installOpenFlowErr: fmt.Errorf("failure to install flow"),
+			expRequeue:         true,
+		}, {
+			name:               "succeeded",
+			installOpenFlowErr: nil,
+			expRequeue:         false,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			controller := gomock.NewController(t)
+			stopCh := make(chan struct{})
+			defer close(stopCh)
+
+			waiter := newAsyncWaiter(testPodNameA, podInfraContainerID, stopCh)
+
+			testClients := newMockClients(controller, nodeName, pod)
+			testClients.startInformers(stopCh)
+			fakeOFClient := testClients.ofClient
+
+			configurator := newTestPodConfigurator(testClients, waiter)
+			defer configurator.unreadyPortQueue.ShutDown()
+
+			configurator.ifaceStore.AddInterface(&interfacestore.InterfaceConfig{
+				InterfaceName: podIfName,
+				ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{
+					PodNamespace: testPodNamespace,
+					PodName:      testPodNameA,
+					ContainerID:  podInfraContainerID,
+				},
+				OVSPortConfig: &interfacestore.OVSPortConfig{
+					PortUUID: "test-port-uuid",
+					OFPort:   int32(1),
+				},
+				IPs: podIPs,
+				MAC: podMac,
+			})
+
+			fakeOFClient.EXPECT().InstallPodFlows(podIfName, podIPs, podMac, portStatusMsg.Desc.PortNo, uint16(0), nil).Times(1).Return(tc.installOpenFlowErr)
+			configurator.unreadyPortQueue.Add(podIfName)
+
+			configurator.processNextWorkItem()
+
+			time.Sleep(600 * time.Millisecond)
```

Comment:
This is not reliable, espeically on windows. And it could extend the execution time 600ms * cases. why don't follow the suggestion that replaces `retryInterval` with a negative number which makes it enqueued immediately?

Reply:
updated.